### ISSUE TITLE
Push queue test spawns orphaned dev_appserver.py processes

### ISSUE THREAD

**Issue by venantius** [golang/appengine#135]
## Summary

This is a separate issue from #134, although I'm running into both in the same codebase at the same time. 

TL;DR - when I try to run a test involving a task getting added to the push queue, the add happens silently (regardless of whether or not the actual path is correct) but a phantom dev_appserver.py process is spawned that doesn't get shut down when the test completes. Eventually, there are so many of these that I start running into SQLite database locking issues and can't run any more tests without manually searching for the orphaned processes and killing them.

I'm not sure if this issue actually belongs here or on another tracker, but since the codebase is in Go and I'm running into it via the Go App Engine SDK I thought I'd start here and you can always route me to a more relevant issue tracker if there is one. 

## Sample Code

app.yaml:
```yaml
runtime: go
api_version: go1

handlers:
 - url: /worker/.*
   script: _go_app
   login: admin
 - url: /.*
   script: _go_app
```

worker/settle.go
```go
package worker

func SettleWorker(w http.ResponseWriter, r *http.Request) {
        ctx := appengine.NewContext(r)
        log.Infof(ctx, "Worker was successfully invoked")
}
```

service/main.go
```go
package service

func TestHandler(w http.ResponseWriter, r *http.Request) {
        ctx := appengine.NewContext(r)
        t := taskqueue.NewPOSTTask("/worker/settle", map[string][]string{"name": {"BLAH"}})
        task, err := taskqueue.Add(ctx, t, "")
        if err != nil {
                http.Error(w, "Error", http.StatusBadRequest)
                return
        }
        log.Infof(ctx, "Successfully posted task to worker: %+v", task)
}

func init() {
        http.HandleFunc("/worker/settle", worker.SettleWorker)
        http.HandleFunc("/test", TestHandler)
}
```

service/main_test.go
```go
package service_test

func TestTestHandler(t *testing.T) {
        inst, err := aetest.NewInstance(nil)
        if err != nil {
                t.Fatal(err.Error())
        }
        req, err := inst.NewRequest("GET", "/", nil)
        if err != nil {
                t.Fatal(err.Error())
        }
        resp := httptest.NewRecorder()
        http.HandlerFunc(service.TestHandler).ServeHTTP(resp, req)
}
```

## Further notes

At a glance, this test seems to causes no issues when run using `goapp test bitwyrm/service -v`. The logging statement from SettleWorker does not appear, but I'm not sure if tasks are meant to get picked up by dev_appserver.py in a test environment (I thought that they were but I can't for the life of me seem to make it work).

This test also leaves an orphaned dev_appserver.py process (regardless of which path is used). Running this repeatedly eventually fills up my computer with orphaned test processes, which eventually causes the dev server invoked via `goapp test` to run into database locking issues, meaning I can't run my test suite without manually killing all of the orphaned test processes. Typical stacktrace:

```
...successful test output...
INFO     2018-05-04 09:02:10,762 stub_util.py:357] Applying all pending transactions and saving the datastore
INFO     2018-05-04 09:02:13,827 stub_util.py:360] Saving search indexes
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/Users/ursa/Code/google-cloud-sdk/platform/google_appengine/google/appengine/api/taskqueue/taskqueue_stub.py", line 2182, in MainLoop
    self._ProcessQueues()
  File "/Users/ursa/Code/google-cloud-sdk/platform/google_appengine/google/appengine/api/taskqueue/taskqueue_stub.py", line 2127, in _ProcessQueues
    response_code = self.task_executor.ExecuteTask(task, queue)
  File "/Users/ursa/Code/google-cloud-sdk/platform/google_appengine/google/appengine/api/taskqueue/taskqueue_stub.py", line 2059, in ExecuteTask
    '0.1.0.2')
  File "/Users/ursa/Code/google-cloud-sdk/platform/google_appengine/google/appengine/tools/devappserver2/dispatcher.py", line 776, in add_request
    inst)
  File "/Users/ursa/Code/google-cloud-sdk/platform/google_appengine/google/appengine/tools/devappserver2/dispatcher.py", line 862, in _handle_request
    request_type=request_type)
  File "/Users/ursa/Code/google-cloud-sdk/platform/google_appengine/google/appengine/tools/devappserver2/module.py", line 818, in _handle_request
    module=self._module_configuration.module_name)
  File "/Users/ursa/Code/google-cloud-sdk/platform/google_appengine/google/appengine/api/apiproxy_stub.py", line 186, in WrappedMethod
    return method(self, *args, **kwargs)
  File "/Users/ursa/Code/google-cloud-sdk/platform/google_appengine/google/appengine/api/logservice/logservice_stub.py", line 181, in start_request
    host, start_time, method, resource, http_version, module))
OperationalError: database is locked
```

Note that this all seems to work fine outside of testing - when I hit this http handler in dev using Paw and dev_appserver.py, the correct path does trigger the worker, and the incorrect path sees logging of 403 responses with a series of retries (both behaviors you'd expect).

**Comment by venantius** [comment]
Closing due to PEBCAK - I forgot to `defer inst.Close()` as part of my test body.
